src/doc/trpl/documentation.md

```diff
@@ -1,224 +1,222 @@
+are three kinds of headers that are commonly used. They aren't special syntax,
+just convention, for now.
 
-~~~md
 ```
-// This is a testable code block
+/// # Panics
+# fn foo() {}
 ```
 
-```rust{.example}
-// This is rust and also testable
-```
+If your function or method can panic, you should note the conditions in which
+it panics here. Any function in Rust can panic, so documentation is extra
+important.
 
-```ignore
-// This is not a testable code block
+```
+/// # Failures
+# fn foo() {}
 ```
 
-    // This is a testable code block (4-space indent)
+If your function or method returns a `Result<T, E>`, then describing the
+conditions under which it returns `Err<E>` is a nice thing to do. This is
+slightly less important than `Panics`, because failure is encoded into the type
+system, but it's still a good thing to do.
 
-```sh
-# this is shell code and not tested
 ```
-~~~
+/// # Safety
+# fn foo() {}
+```
 
-You can specify that the test's execution should fail with the `should_fail`
-directive.
+If your function is `unsafe`, you should explain which invariants the caller is
+responsible for upholding.
 
-~~~md
-```should_fail
-// This code block is expected to generate a panic when run
 ```
-~~~
+/// # Examples
+///
+/// ```
+/// use std::rc::Rc;
+///
+/// let five = Rc::new(5);
+/// ```
+# fn foo() {}
+```
 
-You can specify that the code block should be compiled but not run with the
-`no_run` directive.
+Third, `Examples`. Include one or more examples of using your function or
+method, and your users will love you for it. These examples go inside of
+code block annotations, which we'll talk about in a moment, and can have
+more than one section:
 
-~~~md
-```no_run
-// This code will be compiled but not executed
 ```
-~~~
+/// # Examples
+///
+/// Simple `&str` patterns:
+///
+/// ```
+/// let v: Vec<&str> = "Mary had a little lamb".split(' ').collect();
+/// assert_eq!(v, vec!["Mary", "had", "a", "little", "lamb"]);
+/// ```
+///
+/// More complex patterns with a lambda:
+///
+/// ```
+/// let v: Vec<&str> = "abc1def2ghi".split(|c: char| c.is_numeric()).collect();
+/// assert_eq!(v, vec!["abc", "def", "ghi"]);
+/// ```
+# fn foo() {}
+```
 
-Lastly, you can specify that a code block be compiled as if `--test`
-were passed to the compiler using the `test_harness` directive.
+Let's discuss the details of these code blocks.
 
-~~~md
-```test_harness
-#[test]
-fn foo() {
-    panic!("oops! (will run & register as a failed test)")
-}
-```
-~~~
+#### Code block annotations
 
-Rustdoc also supplies some extra sugar for helping with some tedious
-documentation examples. If a line is prefixed with `# `, then the line
-will not show up in the HTML documentation, but it will be used when
-testing the code block (NB. the space after the `#` is required, so
-that one can still write things like `#[derive(Eq)]`).
+To write some Rust code in a comment, use the triple graves:
```

Comment:
(Prescriptively/pedantically, apparently the use of '`' in programming is technically a backtick or backquote... not necessary to change, though.)